EIPS/eip-7733.md

```diff
@@ -0,0 +1,232 @@
+---
+eip: 7733
+title: overflow protection
+description: zk-SNARK public inputs overflow protection
+author: saya (@0xsaya), Bryce (@brycewai), BeosinTeam (@BeosinTeam)
+discussions-to: https://ethereum-magicians.org/t/eip-zk-snark-public-inputs-overflow-protection/15618
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-20
+requires: 196, 1922
+---
+
+## Abstract
+
+This standard defines that the verify function using the zk-SNARK ("Zero-Knowledge Succinct Non-Interactive Arguments of Knowledge") algorithm needs to check the valid range of the input parameters before use. If the check passes, it returns true and continues to verify the proof. Otherwise, it returns false. The languages involved include: Solidity, JavaScript, Java, Rust, etc.
```

Comment:
```suggestion
This standard mandates that the verify function using the zk-SNARK ("Zero-Knowledge Succinct Non-Interactive Arguments of Knowledge") algorithm needs to check the valid range of the input parameters before use. If the check passes, it returns true and continues to verify the proof. Otherwise, it returns false. The languages involved include: Solidity, JavaScript, Java, Rust, etc.
```